### PR TITLE
Update SQL reference to state that decimal support is currently experimental

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/4036

# Rationale for this change

We don't want users to be surprised

# What changes are included in this PR?

Update SQL reference to state that decimal support is currently experimental

# Are there any user-facing changes?

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Fine with me but I think @liukun4515  should weigh in here prior to merging

File: docs/source/user-guide/sql/data_types.md [hunk 1]
```diff
@@ -35,20 +35,20 @@ This mapping occurs when defining the schema in a `CREATE EXTERNAL TABLE` comman
...
+| `INT UNSIGNED` or `INTEGER UNSIGNED` | `UInt32`          |                                                                                                             |
+| `BIGINT UNSIGNED`                    | `UInt64`          |                                                                                                             |
+| `FLOAT`                              | `Float32`         |                                                                                                             |
+| `REAL`                               | `Float32`         |                                                                                                             |
+| `DOUBLE`                             | `Float64`         |                                                                                                             |
+| `DECIMAL(p,s)`                       | `Decimal128(p,s)` | Decimal support is currently experimental ([#3523](https://github.com/apache/arrow-datafusion/issues/3523)) |
```
Comment (HaoYang670) [inline docs/source/user-guide/sql/data_types.md:51]:
```suggestion
| `DECIMAL(precision,scale)`                       | `Decimal128(precision,scale)` | Decimal support is currently experimental ([#3523](https://github.com/apache/arrow-datafusion/issues/3523)) |
```

**Review by HaoYang670 — APPROVED** [review]
LGTM. Left a small nit.

**Review by liukun4515 — APPROVED** [review]
LGTM

**Comment by alamb** [comment]
I took the liberty of merging from master and running prettier (in [ca79643](https://github.com/apache/arrow-datafusion/pull/4109/commits/ca79643fe8318366c94fc051bbe8a3e4ba36d12c)) so we could get this PR merged and out of the backlog

**Pull request merged**